local-tests/chainrunner/node.py

```diff
@@ -76,14 +80,42 @@ def greplog(self, regexp):
             log = f.read()
         return re.findall(regexp, log)
 
+    def highest_hash(self):
+        highest_block = self.rpc('chain_getBlockHash', None)
+        if isinstance(highest_block, Ok):
+            return highest_block.result
+        else:
+            None
+
+    def highest_finalized_hash(self):
+        highest_finalized = self.rpc('chain_getFinalizedHead', None)
+        if isinstance(highest_finalized, Ok):
+            return highest_finalized.result
+        else:
+            return None
+
+    def block_number(self, hash):
+        block = self.rpc('chain_getBlock', [hash])
+        if isinstance(block, Ok):
+            return int(block.result['block']['header']['number'], 16)
+        else:
+            return -1
+
     def highest_block(self):
-        """Find in the logs the height of the most recent block.
+        """Find the height of the most recent block.
         Return two ints: highest block and highest finalized block."""
-        results = self.greplog(r'best: #(\d+) .+ finalized #(\d+)')
-        if results:
-            a, b = results[-1]
-            return int(a), int(b)
-        return -1, -1
+        try:
+            highest_block_hash = self.highest_hash()
+            highest_finalized_hash = self.highest_finalized_hash()
```

Comment:
we're getting the highest block first and then the highest finalized. When doing things in this order it's possible that highest finalized > highest known block :stuck_out_tongue:. I don't think we use it anywhere so feel free to resolve if it doesn't bother you.